sdk/android/skin/src/main/java/com/ooyala/android/skin/OoyalaSkinLayoutController.java

```diff
@@ -169,33 +153,34 @@ public OoyalaSkinLayoutController(Application app, OoyalaSkinLayout layout, Ooya
     width = Math.round(_layout.getViewWidth() * cal);
     height = Math.round(_layout.getViewHeight() * cal);
 
-
     isReactMounted = false;
 
     initializeSkin(app, layout, player, skinOptions);
   }
 
-  private void initializeSkin(Application app, OoyalaSkinLayout l, OoyalaPlayer p, SkinOptions skinOptions) {
-
+    private void initializeSkin(Application app, OoyalaSkinLayout skinLayout, OoyalaPlayer p, SkinOptions skinOptions) {
+    final Context context = skinLayout.getContext();
     if (skinOptions.getEnableReactJSServer()) {
-      ReactUtil.setJSServer(skinOptions.getReactJSServerHost(), l.getContext());
-    }
-    JSONObject configJson = SkinConfigUtil.loadInitialProperties(l.getContext(), skinOptions.getSkinConfigAssetName());
-    SkinConfigUtil.applySkinOverridesInPlace(configJson, skinOptions.getSkinOverrides());
-    injectLocalizedResources(configJson, l.getContext());
-    saveUpNextSetting(configJson);
-    Bundle launchOptions = null; //Initial properties.
-    if (configJson != null) {
-      try {
-        launchOptions = BundleJSONConverter.convertToBundle(configJson);
-        closedCaptionsSkinStyle = configJson.getJSONObject("closedCaptionOptions");
-        setDefaultAudioLanguage(configJson);
-      } catch (JSONException e) {
-        e.printStackTrace();
-        launchOptions = null;
-      }
+      ReactUtil.setJSServer(skinOptions.getReactJSServerHost(), context);
     }
+    initConfig(context, skinOptions);
+    initSkinViews(app, skinLayout, skinOptions);
+  }
+
+  private void initConfig(Context context, SkinOptions skinOptions) {
+    JSONObject configJson = AssetUtil.loadJsonAsset(context, skinOptions.getSkinConfigAssetName());
+    configManager = new SkinConfigManager(configJson);
+    configManager.removeNullsFromPlaybackArray();
```

Comment:
I don't think it will work for iOS. Those changes should be added to React Native.

Reply:
Alina, please take a look at my comment below.